activation/poetdb.go

```diff
@@ -0,0 +1,129 @@
+package activation
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	xdr "github.com/nullstyle/go-xdr/xdr3"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/types"
+	"github.com/spacemeshos/merkle-tree"
+	"github.com/spacemeshos/poet/hash"
+	"github.com/spacemeshos/poet/shared"
+	"github.com/spacemeshos/poet/verifier"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type PoetDb struct {
+	store database.DB
+}
+
+func NewPoetDb(store database.DB) *PoetDb {
+	return &PoetDb{store: store}
+}
+
+func (db *PoetDb) AddMembershipProof(proof types.PoetMembershipProof) error {
+	root, err := calcRoot(proof.Members)
```

Comment:
I can't think of anything that can go wrong.

`calcRoot` just assumes the slice is a list of leaves of a tree and calculates its root. It handles `nil` (the root will be `nil`) and any other list of values I can think of. If I can't think of a value that would break this, I also can't validate or otherwise handle it. If you have another specific example that I can consider -- lmk.